src/prog_models/models/battery_electrochem.py

```diff
@@ -729,7 +730,7 @@ def __init__(self, **kwargs):
         self.param_callbacks['Ro'] = [OverwrittenWarning]
         super().__init__(**kwargs)
 
-    def dx(self, x : dict, u : dict):
+    def dx(self, x: Union[dict, "StateContainer"], u : dict):
```

Comment:
```suggestion
    def dx(self, x: StateContainer, u : dict):
```

Let's have the type hint only be state container, since that what we want users to use. You can import StateContainer from containers now (as of today). You will need to merge dev first.